test/heapwatch/metrics_delta.py

```diff
@@ -67,74 +193,223 @@ def main():
+                    ip_to_name[ip] = k
+        #logger.debug('names: %r', sorted(ip_to_name.values()))
+        #logger.debug('ip to name %r', ip_to_name)
+        for ip, name in ip_to_name.items():
+            found = []
+            for nick in filesByNick.keys():
+                if ip in nick:
+                    found.append(nick)
+            if len(found) == 1:
+                nick_to_tfname[found[0]] = name
+            elif len(found) > 1:
+                logger.warning('ip %s (%s) found in nicks: %r', ip, name, found)
+            else:
+                logger.warning('ip %s no nick')
+        #logger.debug('nick_to_tfname %r', nick_to_tfname)
+
+    if args.nick_re:
+        # use each --nick-re=foo as a group
+        for nre in args.nick_re:
+            rsum = summary()
+            nretup = (nre,)
+            for rnick, paths in filesByNick.items():
+                nick = nick_to_tfname.get(rnick, rnick)
+                if anynickre(nretup, (rnick,nick)):
+                    rsum(process_files(args, nick, paths), nick)
+            print(rsum)
+            print('\n')
+        return 0
+
+    # no filters, glob it all up
+    rsum = summary()
+    if nonick:
+        rsum(process_files(args, None, nonick), 'no nick')
+    for rnick, paths in filesByNick.items():
+        nick = nick_to_tfname.get(rnick, rnick)
+        rsum(process_files(args, nick, paths), nick)
+    print(rsum)
     return 0
 
+def perProtocol(prefix, lists, sums, deltas, dt):
+    lp = len(prefix)
+    for k, v in deltas.items():
+        if k.startswith(prefix):
+            sub = k[lp:]
+            dapp(lists, sub, v/dt)
+            sums[sub] = sums.get(sub,0) + v
+
+def process_files(args, nick, paths):
+    return nodestats().process_files(args, nick, paths)
+
+class nodestats:
+    def __init__(self):
+        self.nick = None
+        self.args = None
+        self.deltas = []
+        self.txBpsList = []
+        self.rxBpsList = []
+        self.tpsList = []
+        self.txBSum = 0
+        self.rxBSum = 0
+        self.txnSum = 0
+        self.secondsSum = 0
+        # algod_network_received_bytes_*
+        self.rxPLists = {}
+        self.rxPSums = {}
+        # algod_network_sent_bytes_*
+        self.txPLists = {}
+        self.txPSums = {}
+
+    def process_files(self, args, nick=None, metrics_files=None):
+        self.args = args
+        self.nick = nick
+        if metrics_files is None:
+            return self
+        reportf = None
+        writer = None
+        reportpath = None
+        if args.report:
+            if args.report == '-':
+                writer = csv.writer(sys.stdout)
+            else:
+                if nick is None:
+                    reportpath = args.report
+                elif args.report.endswith('.csv'):
+                    reportpath = args.report[:-4] + nick + '.csv'
+                reportf = open(reportpath, 'wt')
```

Comment:
This may accidentally overwrite existing files. 
Why not ask for confirmation to overwrite, append, or write only to new files?

Reply:
for this analysis script I definitely want it to overwrite by default, in that I frequently tweak the script, run it again, and want to see the new reports.